natter-api/src/main/java/com/manning/apisecurityinaction/controllers/TokenController.java

```diff
@@ -28,34 +30,45 @@ public TokenController(TokenStore tokenStore) {
         this.tokenStore = tokenStore;
     }
 
-    public JSONObject login(Request request, Response response) {
-        String subject = request.attribute("subject");
-        var expiry = Instant.now().plus(10, ChronoUnit.MINUTES);
-        var token = new TokenStore.Token(expiry, subject);
-        var tokenId = tokenStore.create(request, token);
 
-        response.status(201);
-        return new JSONObject().put("token", tokenId);
-    }
-    
     public void validateToken(Request request, Response response) {
-        // WARNING: CSRF attack possible!
-        var tokenId = request.headers("X-CSRF-Token");
-        if (tokenId == null) return;
-        
+        var tokenId = request.headers("Authorization");
+        if (tokenId == null || tokenId.startsWith("Bearer ")) {
+            return;
+        }
```

Comment:
confusing name `tokenId` for the var and also duplicated code between `validateToken` and `login`.